src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
+   */
+  [kExportGeoJSONStream]({
+    observations = true,
+    tracks = true,
+    lang,
+    seenAttachments = new Map(),
+  } = {}) {
+    // Format based on https://doc.arcgis.com/en/arcgis-online/reference/geojson.htm
+
+    return Readable.from(
+      this.#exportGeoJSONIterator({
+        observations,
+        tracks,
+        lang,
+        seenAttachments,
+      })
+    )
+  }
+
+  /**
+   *
+   * @param {string} type Type of export this will be
+   * @returns {Promise<string>}
+   */
+  async #exportPrefix(type = '') {
+    const name = await this.#getProjectName()
+    const date = new Date()
+    const dateSection = date
+      .toLocaleDateString('nu', {
+        year: 'numeric',
+        month: 'numeric',
+        day: 'numeric',
+      })
+      .replaceAll('/', '_')
+      .replaceAll('-', '_')
+    return `CoMapeo_${name}_${type}_${dateSection}`
+  }
+
+  /**
+   * @param {boolean} observations
+   * @param {boolean} tracks
+   * @returns {Promise<string>}
+   */
+  async [kGeoJSONFileName](observations, tracks) {
+    let exportType = ''
+    if (observations) exportType += 'Obsvns'
+    if (tracks) {
+      if (observations) exportType += '_'
+      exportType += 'Tracks'
+    }
+    const prefix = await this.#exportPrefix(exportType)
+
+    return prefix + '.geojson'
+  }
+
+  /**
+   * @param {boolean} observations
+   * @param {boolean} tracks
+   * @returns {Promise<string>}
+   */
+  async #zipFileName(observations, tracks) {
+    let exportType = ''
+    if (observations) exportType += 'Obsvns'
+    if (tracks) {
+      if (observations) exportType += '_'
+      exportType += 'Tracks'
+    }
+    const prefix = await this.#exportPrefix(exportType)
+
+    return prefix + '.zip'
+  }
+
+  /**
+   * Export observations and or tracks as a GeoJSON file
+   * @param {string} exportFolder Path to save the file. The file name is auto-generated
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {string} [options.lang]
+   * @returns {Promise<string>} The full path that the file was exported at
+   */
+  async exportGeoJSONFile(
+    exportFolder,
+    { observations = true, tracks = true, lang } = {}
+  ) {
+    const fileName = await this[kGeoJSONFileName](observations, tracks)
+    const filePath = path.join(exportFolder, fileName)
+    const source = this[kExportGeoJSONStream]({ observations, tracks, lang })
+    const sink = createWriteStream(filePath)
+    await pipelinePromise(source, sink)
+
+    return filePath
+  }
+
+  /**
+   * @param {Attachment} attachment
+   * @returns {Promise<null | BlobId>}
+   */
+  async #tryGetBlobId(attachment) {
+    // Audio must not have variants
+    for (const variant of VARIANT_EXPORT_ORDER) {
+      const blobId = buildBlobId(attachment, variant)
```

Comment:
I believe `buildBlobId` will throw if an unknown attachment type is encountered, and this is not caught anywhere and will lead to a failed export. We should export unknown attachment types without error. The user story for this is a new version of that app adds a new attachment type, syncs with a user with an older version of the app, and the user with the older version exports data. They would expect to get all data, including the unknown attachment types, in their export.

Reply:
The current TS types require a variant to be specified in `SUPPORTED_BLOB_VARIANTS`. Should I make it possible to bypass this inside the blob store API?

IMO it's not the end of the world if an old app skips data it doesn't understand.

cc @achou11 on ideas for how to make the TS types happy with unknown variants.

```javascript
const SUPPORTED_BLOB_VARIANTS = /** @type {const} */ ({
  photo: ['original', 'preview', 'thumbnail'],
  audio: ['original'],
  video: ['original'],
})
```

Reply:
Good point and agreed. I think given that we have no immediate plans to add new blob types or variants, it's ok to skip data that is not understood. I think the fix that is needed to stop unknown data throwing, and instead ignore it.

Reply:
Added catch for this